components/brave_rewards/core/zebpay/zebpay_util.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/core/zebpay/zebpay_util.h"
+
+#include "base/base64.h"
+#include "base/strings/escape.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/brave_rewards/core/buildflags.h"
+#include "brave/components/brave_rewards/core/rewards_engine_impl.h"
+
+namespace brave_rewards::internal {
+namespace {
+enum class UrlType { kOAuth, kAPI };
+
+std::string GetUrl(UrlType type) {
+  if (type == UrlType::kOAuth) {
+    return _environment == mojom::Environment::PRODUCTION
+               ? BUILDFLAG(ZEBPAY_PRODUCTION_OAUTH_URL)
+               : BUILDFLAG(ZEBPAY_SANDBOX_OAUTH_URL);
+  } else {
+    DCHECK(type == UrlType::kAPI);
+    return _environment == mojom::Environment::PRODUCTION
+               ? BUILDFLAG(ZEBPAY_PRODUCTION_API_URL)
+               : BUILDFLAG(ZEBPAY_SANDBOX_API_URL);
+  }
+}
+
+std::string GetAccountUrl() {
+  return GetUrl(UrlType::kAPI) + "/dashboard";
+}
+
+std::string GetActivityUrl() {
+  return GetUrl(UrlType::kAPI) + "/activity";
+}
+
+std::string GetLoginUrl(const std::string& state) {
+  return base::StringPrintf("%s/account/login?returnUrl=",
+                            GetUrl(UrlType::kOAuth).c_str()) +
+         base::EscapeQueryParamValue(
+             base::StringPrintf("/connect/authorize/callback"
+                                "?client_id=%s"
+                                "&scope="
+                                "openid "
+                                "profile "
+                                "wallet:transactions:read "
+                                "trade:read "
+                                "trade:create "
+                                "wallet:address:create "
+                                "wallet:address:read "
+                                "wallet:transactions:send"
+                                "&state=%s"
```

Comment:
Do we check that the state we receive in the `rewards://zebpay/authorization` request matches the one we passed here?

Reply:
Yes, we do:
https://github.com/brave/brave-core/blob/760f994002820ae85421061cfac48e987a605c03/components/brave_rewards/core/wallet_provider/connect_external_wallet.cc#L121